### PR TITLE
Co: Simplified configuration

### PR DESCRIPTION
**Pull request by Shine-neko**
| Questions | Answers |
| --- | --- |
| Branch? | develop |
| Description? | Unify two configuration files "settings.in.php" and "parameters.yml,  move all cache files in app/cache/{env} and log in app/logs |
| Type? | Improvement |
| BC breaks? | yes |
| Deprecations? | yes |
| How to test? |  Clear cache ( rm -r app/cache/dev) |

### CODE REVIEW

File: install-dev/controllers/http/process.php [hunk 1]
```diff
@@ -79,7 +79,7 @@ public function process()
         $this->clearConfigXML() && $this->clearConfigThemes();
 
         if (file_exists(_PS_ROOT_DIR_.'/'.self::SETTINGS_FILE)) {
-            require_once _PS_ROOT_DIR_.'/'.self::SETTINGS_FILE;
+            //require_once _PS_ROOT_DIR_.'/'.self::SETTINGS_FILE;
```
Comment (mickaelandrieu) [inline install-dev/controllers/http/process.php]: meh.

File: src/PrestaShopBundle/CacheWarmer/LocalizationCacheWarmer.php [hunk 2]
```diff
@@ -0,0 +1,53 @@
+<?php
+
```
Comment (mickaelandrieu) [inline src/PrestaShopBundle/CacheWarmer/LocalizationCacheWarmer.php]: class header

File: tests/Unit/classes/PrestaShopAutoloadTest.php [hunk 3]
```diff
@@ -29,14 +29,14 @@
...
+class PrestaShopAutoloadTest extends PHPUnit_Framework_TestCase
 {
     private $file_index_content = null;
+    private $file_index = null;
 
     protected function setUp()
     {
-        $this->file_index = _PS_ROOT_DIR_.DIRECTORY_SEPARATOR.PrestaShopAutoload::INDEX_FILE;
-        unlink($this->file_index);
+        $this->file_index = PrestaShopAutoload::getCacheFileIndex();
```
Comment (mickaelandrieu) [inline tests/Unit/classes/PrestaShopAutoloadTest.php]: there is a bug

**Comment by mickaelandrieu** [comment]
Nice idea :) lets wait for tests, comments and pull request description !

File: config/autoload.php [hunk 4]
```diff
@@ -24,6 +24,8 @@
  * International Registered Trademark & Property of PrestaShop SA
  */
 
+define('_PS_VERSION_', '1.7.0.0');
```
Comment (julienbourdeau) [inline config/autoload.php]: Are you sure this should be hardcoded here? This _define_ is generated on installation based one this one:
https://github.com/PrestaShop/PrestaShop/blob/develop/install-dev/install_version.php

File: config/bootstrap.php [hunk 5]
```diff
@@ -27,3 +27,25 @@
 $container_builder = new \PrestaShop\PrestaShop\Core\ContainerBuilder;
 $container = $container_builder->build();
 \PrestaShop\PrestaShop\Adapter\ServiceLocator::setServiceContainerInstance($container);
+
+if (!file_exists(_PS_CACHE_DIR_. 'appParameters.php')) {
+    $yaml = new \Symfony\Component\Yaml\Yaml;
+    $config = $yaml->parse(file_get_contents(__DIR__. '/../app/config/parameters.yml'));
+    file_put_contents(_PS_CACHE_DIR_ .'appParameters.php', '<?php return ' . var_export($config, true). ';');
```
Comment (julienbourdeau) [inline config/bootstrap.php]: :+1:

**Comment by julienbourdeau** [comment]
💣 

Thank you !

**Comment by Shine-neko** [comment]
http://vomzi.com/wp-content/uploads/2016/02/best-like-a-boss-gif-544.gif

**Comment by Shudrum** [comment]
Not sufficiently tested Pull Request.

If you clone a repo from scratch, the installation cannot work.

The `log` folder was created during the installation then tested. But right now, not created but still tested. This is a simple error, but **I'm really worried about the lack of tests** for a such important pull request.

Please fix this as soon as possible, and make way more tests before submitting the pull request, and @julienbourdeau, before merging.

Thank you!

**Comment by Shine-neko** [comment]
Oh dude !!!! SHAME ON ME  see https://github.com/PrestaShop/PrestaShop/pull/5527

**Comment by Shudrum** [comment]
You rock.

**Pull request merged**